timelock-agent/src/main/java/com/palantir/timelock/invariants/NoSimultaneousServiceCheck.java

```diff
@@ -0,0 +1,126 @@
+
+    private static final int REQUIRED_CONSECUTIVE_VIOLATIONS_BEFORE_FAIL = 5;
+    private static final Duration BACKOFF = Duration.ofMillis(1337);
+
+    private final List<TimeLockActivityChecker> timeLockActivityCheckers;
+    private final Consumer<String> failureMechanism;
+    private final ExecutorService executorService;
+
+    @VisibleForTesting
+    NoSimultaneousServiceCheck(
+            List<TimeLockActivityChecker> timeLockActivityCheckers,
+            Consumer<String> failureMechanism,
+            ExecutorService executorService) {
+        this.timeLockActivityCheckers = timeLockActivityCheckers;
+        this.failureMechanism = failureMechanism;
+        this.executorService = executorService;
+    }
+
+    public static NoSimultaneousServiceCheck create(List<TimeLockActivityChecker> timeLockActivityCheckers) {
+        ExecutorService executorService = PTExecutors.newSingleThreadExecutor(
+                PTExecutors.newNamedThreadFactory(false));
+        return new NoSimultaneousServiceCheck(timeLockActivityCheckers,
+                client -> {
+                    // TODO (jkong): Gather confidence and then change to ServerKiller, so that we ACTUALLY shoot
+                    // ourselves in the head.
+                    log.error("We observed that multiple services were consistently serving timestamps, for the"
+                            + " client {}. This is potentially indicative of SEVERE DATA CORRUPTION, and should"
+                            + " never happen in a correct TimeLock implementation. If you see this message, please"
+                            + " check the frequency of leader elections on your stack: if they are very frequent,"
+                            + " consider increasing the leader election timeout. Otherwise, please contact support -"
+                            + " your stack may have been compromised",
+                            SafeArg.of("client", client));
+                },
+                executorService);
+    }
+
+    public void processHealthCheckDigest(HealthCheckDigest digest) {
+        Set<Client> clientsWithMultipleLeaders = digest.statusesToClient().get(TimeLockStatus.MULTIPLE_LEADERS);
+        if (clientsWithMultipleLeaders.isEmpty()) {
+            return;
+        }
+
+        log.info("Clients {} appear to have multiple leaders based on the leader ping health check. Scheduling"
+                + " checks on these specific clients now.", SafeArg.of("clients", clientsWithMultipleLeaders));
+        clientsWithMultipleLeaders.forEach(this::scheduleCheckOnSpecificClient);
+    }
+
+    private void scheduleCheckOnSpecificClient(Client client) {
+        executorService.submit(() -> {
+            try {
+                performCheckOnSpecificClientUnsafe(client);
+            } catch (Exception e) {
+                log.info("No-simultaneous service check failed, suppressing exception to allow future checks", e);
```

Comment:
Is this actually useful? We don't need to suppress because there are not scheduled. Also if something went wrong with TACs but we keep trying to check, we will never really figure out something is wrong